### DISCUSSION TITLE
Properly expose Immersive Mode on android

### DISCUSSION THREAD

**Discussion by syntaxerror247** [godotengine/godot-proposals#11011] [Platforms]
For a long time, I wasn't aware that Android had a runtime method to toggle immersive mode. It wasn't until a few days ago, when I started working on [this proposal](https://github.com/godotengine/godot-proposals/issues/8826), that I discovered it could already be done using `window_set_mode`.

I have two potential solutions now:

1. Update the documentation to clarify that `immersive mode` is activated when either WINDOW_MODE_EXCLUSIVE_FULLSCREEN or WINDOW_MODE_FULLSCREEN is set.
2. Introduce a new method in `DisplayServer`. We could keep `window_set_mode` as it is and add a dedicated `toggle_immersive_mode()` function for better clarity and control.

Let me know which solution you think is better :)

**Comment by m4gr3d** [comment]
Let's go with the first approach! 
The use of `window_set_mode` also provides us with added flexibility given that pure immersive mode is not available on all platforms.

**Comment by syntaxerror247** [comment]
@m4gr3d Ok, so if we are going with first approach, then I would like to add a note in android export `Immersive mode` toggle description : "See `window_set_mode` to toggle it at runtime".

**Reply by m4gr3d** [reply]
Yeah, that's a good idea! I was debating removing or at least renaming that export setting because the ability to do so at runtime should supercede it.
The only advantage of that setting now is to activate fullscreen before the game logic is able to run.

**Reply by syntaxerror247** [reply]
I am not sure if removing would be a good idea because games mostly use immersive mode and setting it during build is better.
But yeah renaming should be considered.
